### ISSUE TITLE
Closing Komodo IDE using red “X” in upper right causes unclean shutdown

### ISSUE THREAD

**Issue by mjross** [Komodo/KomodoEdit#3750]
### Summary
For apparently multiple users (but not all), when Komodo IDE is closed using the red “X” button in the upper right corner, then when Komodo is next started, it immediately presents a pop-up message stating that Komodo was not shutdown cleanly the last time it ran. When Komodo IDE is closed using File > Exit (or a key sequence assigned to it), the pop-up message never appears

### Steps to Reproduce
Help > Troubleshooting > Restart in Safe-Mode. Close Komodo using the red “X” button in the upper right corner. Restart Komodo normally.

### Expected Results
Nothing unusual.

### Actual Results
A pop-up message stating that Komodo was not shutdown cleanly the last time it ran.

### Platform Information
Komodo IDE 11.1.1 build 91089 (on Windows Home 10.0.17763.437)

### Attachments
![restart_Komodo_after_red_x](https://user-images.githubusercontent.com/829492/57255742-65269a00-701a-11e9-8059-e3c9f2983a39.jpg)

**Comment by th3coop** [comment]
If anyone notices additional steps that leads to this that would be helpful. So far, starting Komodo then closing right away using `x` in top right does no reproduce.

**Comment by trogne** [comment]
I have the same issue, Windows 10, Komodo 11.1.2.   Thanks for opening this issue.

**Comment by mjross** [comment]
@trogne Certainly.

**Comment by th3coop** [comment]
> If anyone notices additional steps that leads to this that would be helpful

@mjross  and @trogne, please keep ^that^ in mind as you use Komodo.  

Also, I wanted to clarify, if you follow the steps I mentioned, you DO or DO NOT hit the issue?  ie. Start Komodo and immediately close it using the red `x`.

**Comment by mjross** [comment]
Yes, I do hit the issue. I tried it again just now. I closed Komodo using File > Exit, immediately restarted it (no error), immediately closed it using the red 'x', immediately restarted it (got the error), immediately closed it using the red 'x' a second time, and immediately restarted it (got the error again).

**Comment by mjross** [comment]
Here are some log files at each step...

[1_pystderr_before_Komodo_start_first.log.txt](https://github.com/Komodo/KomodoEdit/files/3158963/1_pystderr_before_Komodo_start_first.log.txt)

[2_pystderr_after_Komodo_stop_with_red_x.log.txt](https://github.com/Komodo/KomodoEdit/files/3158964/2_pystderr_after_Komodo_stop_with_red_x.log.txt)

[3_pystderr_after_Komodo_restart.log.txt](https://github.com/Komodo/KomodoEdit/files/3158965/3_pystderr_after_Komodo_restart.log.txt)

**Comment by th3coop** [comment]
Ok thank you, and one last thing: What if you do it in safe mode?

**Comment by mjross** [comment]
Then it doesn't happen. In fact, for that reason, I had to update my original post to indicate "Start Komodo normally (i.e., not in Safe-Mode)." I'm guessing Safe-Mode wipes the slate clean.

**Comment by th3coop** [comment]
A clue!  Finally.  Could you zip up your profile folder, without any codeintel related files, and send it to Support@activestate.com saying it's for me and in reference to this ticket?

**Comment by mjross** [comment]
I zipped them up, but Gmail won't let me send them ("Blocked for security reasons!"). So here they are: https://send.firefox.com/download/2568397fc39ce6b9/#k6cmrGeOVaxzNuDfx-qGWw

Even though I closed Komodo beforehand, 7-Zip detected two files in use and apparently didn't include them.

**Comment by th3coop** [comment]
@mjross that link expired.  Sorry, we had something going on a work that took my full attention last week.  Were the files that weren't included Code Intel related by any chance?  I recall from the forum post that you two, or one of you, was often noticing that Komodo wasn't stopping when you closed it.  This is often due to code intel taking a long time to shutdown.   I wonder if a code intel reset fixes this.

**Comment by mjross** [comment]
@th3coop No worries; here's a fresh one, with codeintel-related files in case they are helpful: https://send.firefox.com/download/b51c632dd34baa36/#_eu8lxA6Qwfsj9_hl4Ebyw. Per your instructions, I hadn't included any codeintel related files in that previous one. I haven't had the experience of Komodo actually not stopping (unless I'm forgetting something from awhile back). Let me know if I can send along anything else or try a test of some sort.

**Comment by th3coop** [comment]
@mjross, no dice but there are a lot of missing files from the profile.  Are you sure you zipped everything up?  There's no tools folder, prefs.xml, and others.

**Comment by mjross** [comment]
Yes. I opened the file just now and I see all that you mentioned. Image below...
![Clipboard01](https://user-images.githubusercontent.com/829492/58135290-7ad9b900-7bef-11e9-9ff7-a9f6294f4277.jpg)

**Comment by th3coop** [comment]
gah sorry for the long delay.  The link has now expired.  Any chance you still have the file available to share @mjross ?

**Comment by mjross** [comment]
@th3coop No, it's long gone, so I re-created it. Here's the link: https://send.firefox.com/download/30cc134c57282854/#COyGDnkevpmYaX7HkqHkZA. Please download it ASAP. Thanks.

**Comment by th3coop** [comment]
@mjross thank you.
